src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/instance_trial_of_the_crusader.cpp

```diff
@@ -153,9 +158,22 @@ class instance_trial_of_the_crusader : public InstanceMapScript
             void OnCreatureCreate(Creature* creature) override
             {
                 InstanceScript::OnCreatureCreate(creature);
-                if (creature->GetEntry() == NPC_BARRET_RAMSEY)
-                    if (!TrialCounter)
-                        creature->DespawnOrUnsummon();
+                switch (creature->GetEntry())
+                {
+                    case NPC_BARRETT_BEASTS:
+                    case NPC_BARRETT_BEASTS_HC:
+                    case NPC_BARRETT_JARAXXUS:
+                    case NPC_BARRETT_FACTION:
+                    case NPC_BARRETT_VALKYR:
+                    case NPC_BARRETT_LK:
+                    case NPC_ANUBARAK:
+                        if (!TrialCounter)
+                            creature->DespawnOrUnsummon();
```

Comment:
you could try InstanceScript::GetCreatureEntry to prevent spawn of these npcs

https://github.com/TrinityCore/TrinityCore/blob/master/src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp#L360-L413